working/declaring-constructors/feature-specification.md

```diff
@@ -329,12 +439,79 @@ It is a compile-time error if:
 
 *   A field parameter is named `_`. *We could allow this but... why?*
 
+### Private field parameters and initializing formals
+
+An identifier is a *private name* if it starts with an underscore (`_`),
+otherwise it's a *public name*.
+
+A private name may have a *corresponding public name*. If the characters of the
+identifier with the leading underscore removed form a valid identifier and a
+public name, then that is the private name's corresponding public name. *For
+example, the corresponding public name of `_foo` is `foo`.* If removing the
+underscore does not leave something which is is a valid identifier *(as in `_`
+or `_2x`)* or leaves another private name *(as in `__x`)*, then the private name
+has no corresponding public name.
+
+The private declared name, *p*, of an initializing formal or field parameter in
+constructor C has a corresponding *non-conflicting public name* if it has a
+corresponding public name, *n*, and no other parameter of the same constructor
+declaration has either of the names *p* or *n* as declared name. *In other
+words, if removing the `_` leads to a collision with another parameter, then
+there is a conflict.*
+
+Given an initializing formal or field parameter with private name *p*:
+
+*   If *p* has a non-conflicting public name *n*, then:
+
+    *   The name of the parameter in C is *n*. *If the parameter is named, this
+        then avoids the compile-time error that would otherwise be reported for
+        a private named parameter.*
+
+    *   The local variable in the initializer list scope of C is *p*. *Inside
+        the body of the constructor, uses of *p* refer to the field, not the
+        parameter.*
+
+    *   If the parameter is an initializing formal, then it initializes a
+        corresponding field with name *p*.
+
+    *   Else the field parameter induces an instance field with name *n*.
+
+    *Any generated API documentation for the parameter should also use *n*.*
+
+*   Else (there is no non-conflicting public name), the name of the parameter is
+    left alone and also used for the initialized or induced field. *If the
+    parameter is named, this is a compile-time error.*
+
+*For example:*
+
+```dart
+class Id {
+  late final int _region = 0;
+
+  this({this._region, final int _value});
```

Comment:
Can all add a use of the initializer-list variable, like:
```dart
  this({this._region, final int _value}) : assert(_region > 0 && _value > 0);
```
to show that the local variable is *p*.